packages/core/src/external/commonwell/document/document-downloader-local.ts

```diff
@@ -50,13 +51,34 @@ export class DocumentDownloaderLocal extends DocumentDownloader {
     const onEnd = () => {
       console.log("Finished downloading document");
     };
-    const downloadResult = await this.downloadFromCommonwellIntoS3(
-      document,
-      fileInfo,
-      onData,
-      onEnd
-    );
+    let downloadResult = await this.downloadFromCommonwellIntoS3(document, fileInfo, onData, onEnd);
 
+    // Check if the detected file type is in the accepted content types
+    if (!isContentTypeAccepted(document.mimeType)) {
+      // If not, update the content type in S3
+      const [detectedFileType, detectedExtension] = detectFileType(Buffer.from(downloadedDocument));
+      console.log(
+        `Updating content type in S3 ${fileInfo.name} from previous mimeType: ${document.mimeType}
+         to detected mimeType ${detectedFileType} and ${detectedExtension}`
+      );
+      const newKey = await this.s3Utils.updateContentTypeInS3(
+        downloadResult.bucket,
+        downloadResult.key,
+        detectedFileType,
+        detectedExtension
+      );
+      const newLocation = downloadResult.location.replace(`${downloadResult.key}`, `${newKey}`);
+      const fileDetailsUpdated = await this.s3Utils.getFileInfoFromS3(
+        newKey,
+        downloadResult.bucket
+      );
+      downloadResult = {
+        ...downloadResult,
+        ...fileDetailsUpdated,
+        key: newKey,
+        location: newLocation,
+      };
```

Comment:
NABD at this point, but this could be moved to a function to help keep the original code easier to read.